litgpt/lora.py

```diff
@@ -156,16 +156,10 @@ def merge(self) -> None:
                 # assign updated weights and quantize by moving to CUDA device
                 self.linear.weight = bnb.nn.Params4bit(weight_data, requires_grad=False, **weight.__dict__)
                 self.linear.weight.cuda(weight.device)
-            # if the pretrained weights and LoRA weights are of compatible dtypes - simply sum them
-            elif torch.finfo(pretrained_dtype).max >= torch.finfo(lora_data.dtype).max:
+            else:
                 # self.linear might be on CPU and lora_data on CUDA
+                # the inplace add will preserve the dtype of linear.weight
```

Comment:
Why is this correct though?
If I finetune with precision="bf16-true", that means the pretrained weights were converted to bfloat16 during training. Why is it correct to then merge weights in the float16 dtype. The range of these dtypes is different.

IMO we should use what we did during training.

Reply:
Opened https://github.com/Lightning-AI/litgpt/pull/1250 with the other fix to the problem